src/compiler/es-next.js

```diff
@@ -0,0 +1,145 @@
+    }
+
+    static _getBabelOptions (filename) {
+        // NOTE: lazy load heavy dependencies
+        var presetStage2     = require('babel-preset-stage-2');
+        var transformRuntime = require('babel-plugin-transform-runtime');
+        var presetES2015     = NODE_VER < 4 ?
+                               require('babel-preset-es2015-loose') :
+                               require('babel-preset-es2015-node4');
+
+        return {
+            presets:             [presetStage2, presetES2015],
+            plugins:             [transformRuntime],
+            filename:            filename,
+            sourceMaps:          true,
+            ast:                 false,
+            babelrc:             false,
+            highlightCode:       false,
+            resolveModuleSource: source => source === 'testcafe' ? COMMON_API_PATH : source
+        };
+    }
+
+    static _isNodeModulesDep (filename) {
+        return relative(CWD, filename)
+                   .split(pathSep)
+                   .indexOf('node_modules') >= 0;
+    }
+
+    _setupSourceMapsSupport () {
+        sourceMapSupport.install({
+            handleUncaughtExceptions: false,
+
+            retrieveSourceMap: filename => {
+                var map = this.sourceMaps[filename];
+
+                if (map)
+                    return { url: filename, map };
+            }
+        });
+    }
+
+    _compileES (code, filename) {
+        // NOTE: lazy load heavy dependencies
+        var babel = require('babel-core');
+
+        if (this.cache[filename])
+            return this.cache[filename];
+
+        var opts     = ESNextCompiler._getBabelOptions(filename);
+        var compiled = babel.transform(code, opts);
+
+        this.cache[filename]      = compiled.code;
+        this.sourceMaps[filename] = compiled.map;
+
+        return compiled.code;
+    }
+
+    _setupRequireHook (globals) {
+        var origRequireExtension = require.extensions['.js'];
+
+        require.extensions['.js'] = (mod, filename) => {
+            // NOTE: remove global API so it's will be unavailable for dependencies
+            globals.remove();
+
+            if (ESNextCompiler._isNodeModulesDep(filename))
+                origRequireExtension(mod, filename);
+            else {
+                var code         = readFileSync(filename);
+                var compiledCode = this._compileES(stripBom(code), filename);
+
+                mod.paths = ESNextCompiler._getNodeModulesLookupPath(filename);
+
+                mod._compile(compiledCode, filename);
+            }
+
+            globals.setup();
+        };
+
+        return origRequireExtension;
+    }
+
+    canCompile (code, filename) {
+        return /\.js$/.test(filename) &&
+               FIXTURE_RE.test(code) &&
+               TEST_RE.test(code);
+    }
+
+    compile (code, filename) {
+        var compiledCode = this._compileES(code, filename);
+        var mod          = new Module(filename, module.parent);
```

Comment:
Nope, this is a part of the deprecated API: https://nodejs.org/api/globals.html#globals_require_extensions

It was used by `require.extensions` and currently appear in many popular packages across npm. However, we shouldn't worry about it, because:
> Since the Module system is locked, this feature will probably never go away. However, it may have subtle bugs and complexities that are best left untouched.